### PR TITLE
feat: Add customAllocationPriorityFixedPrice for productId 186

### PR DESCRIPTION
**Pull request by rackstar**
## Context

Relates to https://github.com/NexusMutual/sdk/issues/153#issue-2349202919

DeltaPrime (UnoRe):
* productId 186
* fixed price private product
* allowedPools in order of priority: `[18, 22, 1]`

## Changes proposed in this pull request

* add customAllocationPriorityFixedPrice function
* add DeltaPrime (UnoRe) / productId 186 custom allocation pool priority
* unit tests

## Test plan

* add unit tests

## Checklist

- [x] Rebased the base branch
- [x] Attached corresponding Github issue
- [x] Prefixed the name with the type of change (i.e. feat, chore, test)
- [x] Performed a self-review of my own code
- [x] Followed the style guidelines of this project
- [x] Made corresponding changes to the documentation
- [x] Didn't generate new warnings
- [x] Didn't generate failures on existing tests
- [x] Added tests that prove my fix is effective or that my feature works

## Review

When reviewing a PR, please indicate intention in comments using the following emojis:
* :cake: = Nice to have but not essential.
* :bulb: = Suggestion or a comment based on personal opinion
* :hammer: = I believe this should be changed.
* :thinking: = I don’t understand something, do you mind giving me more context?
* :rocket: = Feedback

### CODE REVIEW

File: src/lib/quoteEngine.js [hunk 1]
```diff
@@ -256,7 +289,11 @@ const quoteEngine = (store, productId, amount, period, coverAsset) => {
     };
   });
 
-  const allocations = calculateOptimalPoolAllocation(amountToAllocate, poolsData, product.useFixedPrice);
+  // Use a custom pool allocation priority for DeltaPrime (UnoRe)
+  const allocations =
+    productId === DELTA_PRIME_UNO_RE_PRODUCT_ID
```
Comment (MilGard91) [inline src/lib/quoteEngine.js:294]: We should maybe add property in store something like 
```
productPriorityPools: {[productId]: [...prorityPoolOrder]}
```
 just pass them from there. So next product just expand the object. Also don't forget to add them to the cache file so they don't get overriden

**Review by MilGard91 — COMMENTED** [review]
Small suggestion to the solution

File: src/lib/quoteEngine.js (hunk 1 above)
Comment (rackstar) [reply src/lib/quoteEngine.js:295]: good idea 👍

File: src/lib/quoteEngine.js [hunk 2]
```diff
@@ -185,6 +191,37 @@ const calculateOptimalPoolAllocation = (coverAmount, pools, useFixedPrice = fals
...
+  let coverAmountLeft = amountToAllocate;
+
+  while (coverAmountLeft > 0 && customPoolIdPriority.length > 0) {
+    const poolId = customPoolIdPriority.shift();
+    const pool = poolsData.find(poolData => poolData.poolId === poolId);
+
+    const capacityLeft = pool.totalCapacity.sub(pool.initialCapacityUsed);
+    const poolAllocation = bnMin(capacityLeft, coverAmountLeft);
+
+    allocations[poolId] = allocations[poolId] ? allocations[poolId].add(poolAllocation) : poolAllocation;
```
Comment (MilGard91) [inline src/lib/quoteEngine.js:213]: nit: you don't need this check, since it will never go through the same pool more than once

**Review by MilGard91 — COMMENTED** [review]
One comment, everything else looks good

**Review by MilGard91 — APPROVED** [review]
LGTM

**Pull request merged**